0038-image-signature-verification/README.md

```diff
@@ -0,0 +1,287 @@
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+      - [Unit tests](#unit-tests)
+      - [e2e tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Summary
+
+This ZEP proposes adding image signature verification capabilities to Zarf for `images` that are signed with Cosign signatures. The proposal introduces a configurable mechanism that allows package creators to define registry-specific verification rules using glob pattern matching. This enhancement would enable Zarf to verify the authenticity and integrity of container images during package creation by checking their cryptographic signatures against specified certificates or public keys.
+
+By implementing this feature, Zarf would provide stronger security guarantees for air-gapped environments where supply chain security is critical and would provide a mechanism to catch issues early in the package development process. The proposal includes support for both public key infrastructure and keyless verification methods, accommodating various security postures and compliance requirements across different operational environments.
+
+## Motivation
+
+Container image signature verification is a critical component of supply chain security. In air-gapped environments where Zarf deploys packages, ensuring the integrity and authenticity of container images is particularly important since traditional online verification methods are unavailable. Currently, Zarf lacks native support for verifying signatures on container images, which creates a security gap that must be filled with other tools.  Configuring those tools in the airgap (such as Kyverno) can also be complex because the tools must be configured to work offline - configuring tools like cosign manually on package create can also be cumbersome due to the number of images some zarf packages contain.  This feature would also help align with industry best practices for container security as outlined by organizations like the Cloud Native Computing Foundation (CNCF) and the Open Source Security Foundation (OpenSSF).
+
+### Goals
+
+- Implement container image signature verification for `images` being pulled into Zarf packages
+- Provide a flexible configuration mechanism that supports each of the various signing methods supported by Cosign
+
+### Non-Goals
+
+- Support other signature verification systems beyond Cosign (such as Notary or Docker Content Trust)
+- Automatic signature discovery or verification
```

Comment:
To clarify on thee `Automatic signature discovery` being a non-goal, that is specifically in terms of automating "what key/method was used to sign these images"? Based on the examples provided below it seems like discovery/pulling of cosign signatures from an OCI registry _IS_ automated? Essentially trying to understand if this proposal would require a zarf package creator to add `registry1.dso.mil/ironbank/opensource/zarf-dev/zarf/zarf-agent:sha256-91dc9dda4834dd27185af77efcaa2753ae5bd9e4634ac61db0d19624b6d1beb7.sig` to their package images, or not?

Reply:
correct the signatures can usually be looked up as normal through cosign given the right options. in this case I moreso meant trying to validate those signatures on our own without additional configuration from the user (potentially with a list of "known" configurations since registry1, k8s.io, etc would be common images to validate with common configs)

Reply:
So to answer your last question, the user would not need to add that signature to the package for image verification - they are free to include that in their package though if they wish to bring the signature with them to the airgap (and Zarf can help them find the signature still with `zarf dev find-images`)

Reply:
Perfect - that was my assumption and very in favor of this approach 😀